src/engine/renderer/tr_shade.cpp

```diff
@@ -2437,7 +2531,30 @@ static void Render_liquid( int stage )
 	// depth texture
 	GL_BindToTMU( 2, tr.currentDepthImage );
 
-	if ( normalMapping || hasHeightMapInNormalMap )
+	// bind u_HeightMap u_depthScale u_parallaxOffsetBias
+	if ( parallaxMapping )
```

Comment:
reflection requires heightmap or you can get errors similars to that: https://github.com/DaemonEngine/Daemon/issues/246
cubemap reflections has to be merged with lightmap shader to avoid some bugs, and I have some branch sleeping with that improvements.

liquid shader currently does not grok heightmap yet, but this is just a matter of time before they will.

Also, I have plan to merge the liquid shader with the lightmapping one, people would be able to do that:

```
textures/water/water
{
	qer_editorImage textures/water/water_p
	{
		diffuseMap textures/water/water_d
		liquidMap textures/water/water_n
		materialMap textures/water/water_d

		scroll time * .1, time * .1
		scale .5, .5
		refractionIndex 1.3
		fresnelPower 2
		fresnelScale .85
		fresnelBias .05

		blend blend
	}
}
```

The code would store normalmap in normalmap either the keyword is normalMap or liquidMap but liquidMap would set a boolean to true.
In case that boolean would be true, if r_normalMapping would be enabled the liquid computation would be done, else the diffuse one would be displayed as a fallback, reusing the normal map the classic way.

Basically I'm in a process of boilerplating the code before factorizing it.